examples/gateway/azure_openai.yaml

```diff
@@ -0,0 +1,36 @@
+routes:
+  - name: chat
+    route_type: llm/v1/chat
+    model:
+      provider: openai
+      name: gpt-3.5-turbo
+      config:
+        openai_api_type: "azure"
+        openai_api_key: $OPENAI_API_KEY
+        openai_deployment_name: "{your_deployment_name}"
+        openai_api_base: "https://{your_resource_name}-azureopenai.openai.azure.com/"
+        openai_api_version: "2023-05-15"
+
+  - name: completions
+    route_type: llm/v1/completions
+    model:
+      provider: openai
+      name: gpt-3.5-turbo
+      config:
+        openai_api_type: "azuread"
+        openai_api_key: $AZURE_AAD_TOKEN
+        openai_deployment_name: "{your_deployment_name}"
+        openai_api_base: "https://{your_resource_name}-azureopenai.openai.azure.com/"
+        openai_api_version: "2023-05-15"
+
+  - name: embeddings
+    route_type: llm/v1/embeddings
+    model:
+      provider: openai
+      name: text-embedding-ada-002
+      config:
+        openai_api_type: "azure"
+        openai_api_key: $OPENAI_API_KEY
+        openai_deployment_name: "{your_deployment_name}"
+        openai_api_base: "https://{your_resource_name}-azureopenai.openai.azure.com/"
```

Comment:
```suggestion
        openai_api_base: "https://{your_resource_name}-azureopenai.openai.azure.com/deployments/{your_deployment_name}"
```

What would happen if we include `your_deployment_name` in `openai_api_base`?

Reply:
See examples: API key auth: https://github.com/Azure/openai-samples/blob/main/Basic_Samples/Completions/basic_completions_example_restapi.ipynb 

AAD auth: https://github.com/Azure/openai-samples/blob/main/Basic_Samples/AAD_Integration/aad_integration_example_restapi.ipynb

Reply:
Got it, if openai isolate it, I think we should isolate it.